### PR TITLE
Refactor add/remove/list locations in language

### PR DESCRIPTION
**Pull request by zkhin**
┆Issue is synchronized with this [Monday item](https://seed-company-squad.monday.com/boards/3451697530/pulses/3655346047) by [Unito](https://www.unito.io)

### CODE REVIEW

File: src/components/language/language.service.ts [hunk 1]
```diff
@@ -555,8 +536,8 @@ export class LanguageService {
       items: items,
       total: items.length,
       hasMore: false,
-      canCreate: !!permission?.canReadLocationCreate,
-      canRead: !!permission?.canReadLocationRead,
+      canCreate: canCreateLocation,
+      canRead: canCreateLocation,
```
Comment (michaelmarshall) [inline src/components/language/language.service.ts:540]: @CarsonF are you fine with doing it like this for now?

**Review by michaelmarshall — APPROVED** [review]

File: src/components/language/language.service.ts (hunk 1 above)
Comment (CarsonF) [reply src/components/language/language.service.ts:540]: Mmm I don't think so. I think it's important for most people to be able to read and I imagine not many people will be able to create/edit these locations

Comment (zkhin) [reply src/components/language/language.service.ts:540]: @michaelmarshall @CarsonF so can I just set the `canRead` to `true` and remove the `CreateLocation` power while creating a user? and the `CreateLocation` power can be granted from root user like `Organization` and `Language`.

Comment (CarsonF) [reply src/components/language/language.service.ts:540]: Hmm `canCreate` really means `canAdd` in this case. "Can the current user add a location to this specific language?"

The `CreateLocation` power is completely separate. That should only apply when you are creating a location.

Comment (michaelmarshall) [reply src/components/language/language.service.ts:540]: @CarsonF @sethmcknight We need the business rule to determine if a user can add a location to a language. Thoughts?

Comment (CarsonF) [reply src/components/language/language.service.ts:540]: Well maybe. "Business rule" is really vague. Typically when I've said it in the past I've meant event handlers that have "reaction" logic - like syncing budget records. In my mind this just needs to be in the authorization system some how.

**Comment by CarsonF** [comment]
@zkhin I still think checking the "create location" power here is wrong, as it's a completely separate action. Maybe `canCreate` should just be hardcoded to true for now as well?

**Comment by zkhin** [comment]
> @zkhin I still think checking the "create location" power here is wrong, as it's a completely separate action. Maybe `canCreate` should just be hardcoded to true for now as well?

Okay, I will

**Review by CarsonF — APPROVED** [review]

**Comment by CarsonF** [comment]
@zkhin You're adding read/write permissions to `locations` property for a bunch of roles. Can those not be referenced/used instead of the hardcoded true?

(sorry for the back and forth)

**Comment by zkhin** [comment]
> @zkhin You're adding read/write permissions to `locations` property for a bunch of roles. Can those not be referenced/used instead of the hardcoded true?
> 
> (sorry for the back and forth)

Yes, it could be used for `canRead`/`canCreate` in the future but for that I need to do a KT with Mike

**Pull request merged**